src/rpc/blockchain.cpp

```diff
@@ -1490,14 +1490,73 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
     ret.pushKV("mempoolminfee", ValueFromAmount(std::max(pool.GetMinFee(maxmempool), ::minRelayTxFee).GetFeePerK()));
     ret.pushKV("minrelaytxfee", ValueFromAmount(::minRelayTxFee.GetFeePerK()));
 
+    if (with_fee_histogram) {
+        /* TODO: define log scale formular for dynamically creating the
+         * feelimits but with the property of not constantly changing
+         * (and thus screw up client implementations) */
+        static const std::vector<CAmount> feelimits{1, 2, 3, 4, 5, 6, 7, 8, 10,
+            12, 14, 17, 20, 25, 30, 40, 50, 60, 70, 80, 100,
+            120, 140, 170, 200, 250, 300, 400, 500, 600, 700, 800, 1000,
+            1200, 1400, 1700, 2000, 2500, 3000, 4000, 5000, 6000, 7000, 8000, 10000};
+
+        /* keep histogram per...
+         * ... cumulated tx sizes
+         * ... txns (count)
+         * ... cumulated fees */
+        std::vector<uint64_t> sizes(feelimits.size(), 0);
+        std::vector<uint64_t> count(feelimits.size(), 0);
+        std::vector<uint64_t> fees(feelimits.size(), 0);
+
+        for (const CTxMemPoolEntry& e : pool.mapTx) {
+            int size = (int)e.GetTxSize();
+            CAmount fee = e.GetFee();
+            uint64_t asize = e.GetSizeWithAncestors();
+            CAmount afees = e.GetModFeesWithAncestors();
+            uint64_t dsize = e.GetSizeWithDescendants();
+            CAmount dfees = e.GetModFeesWithDescendants();
+
+            CAmount fpb = fee / size; //fee per byte
+            CAmount afpb = afees / asize; //fee per byte including ancestors
+            CAmount dfpb = dfees / dsize; //fee per byte including descendants
+            CAmount tfpb = (afees + dfees - fee) / (asize + dsize - size);
+            CAmount feeperbyte = std::max(std::min(dfpb, tfpb), std::min(fpb, afpb));
+
+            // distribute feerates into feelimits
+            for (size_t i = 0; i < feelimits.size(); i++) {
+                if (feeperbyte >= feelimits[i] && (i == feelimits.size() - 1 || feeperbyte < feelimits[i + 1])) {
+                    sizes[i] += size;
+                    count[i]++;
+                    fees[i] += fee;
+                    break;
+                }
+            }
+        }
+        CAmount total_fees = 0; //track total amount of available fees in mempool
+        UniValue info(UniValue::VOBJ);
```

Comment:
JSON objects are unordered collections, so maybe using an array for "fee_histogram" would make more sense since it would always stay sorted.